### PR TITLE
feat(server): Prisma

### LINKED ISSUE #32: Database SQL Injection

### LINKED ISSUE #57: Prisma Integration

### PR DESCRIPTION
**Pull request by awave1**
This PR adds [Prisma ](prisma.io)integration. Merging into docker branch, because will need to test it a bit more

As a result, `server/src/db` was removed, since that setup is no longer necessary.

This closes #57, closes #32

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -13,27 +13,27 @@ function App () {
...
+          });
         }
-      })
-      .catch((err: Error) => {
+      } catch (err) {
         errorHandler(err);
-      });
+        console.error(err);
+      }
+    }
```
Comment (awave1) [inline client/src/App.tsx:34]: I was trying to follow the auth process and cleaned up the promises here

File: server/prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+model user {
+  userid           String      @id
+  name             String?
+  curator_settings String?
+  access_token     String?
+  refresh_token    String?
+  profile_pic      String?
+  followers        Int?
+  playlists        playlists[]
+}
```
Comment (awave1) [inline server/prisma/schema.prisma:41]: all of this generated by prisma -- [`prisma introspect`](https://www.prisma.io/docs/concepts/components/introspection)

File: server/src/routes/users.ts [hunk 3]
```diff
@@ -49,36 +49,6 @@ app.get('/getAllPlaylists', (req, res) => {
...
-    .then((rtn) => {
-      res.send(rtn);
-    })
-    .catch((err) => {
-      console.error(err);
-      res.status(500);
-      res.send({ error: err });
-    });
-});
-
```
Comment (awave1) [inline server/src/routes/users.ts:81]: these weren't used

File: server/src/actions/prisma.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { PrismaClient } from '@prisma/client';
+
+/**
+ * Singleton instance for `@prisma/client`
+ */
+export class Prisma {
```
Comment (awave1) [inline server/src/actions/prisma.ts:6]: simple singleton class, to make sure we don't create multiple instances of `PrismaClient`

**Comment by natekchua** [comment]
This is some slick shit man, good work!

**Review by natekchua — APPROVED** [review]

**Review by raksdewji — APPROVED** [review]
Looks good!

**Pull request merged**